hack/update-vendor.sh

```diff
@@ -0,0 +1,107 @@
+#!/usr/bin/env bash
+
+# Copyright 2016 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+KUBE_ROOT=$(dirname "${BASH_SOURCE}")/..
+source "${KUBE_ROOT}/hack/lib/init.sh"
+
+kube::log::status "Ensuring prereqs"
+kube::util::ensure_single_dir_gopath
+kube::util::ensure_no_staging_repos_in_gopath
+
+rm -fr vendor
+
+# generate empty go.mod files for staging dirs
+for repo in $(ls staging/src/k8s.io); do
+  echo "module k8s.io/${repo}
+
+go 1.12
```

Comment:
rather than hardcode this - should we be using something more common?  Or does this version have consequences later?

E.g. When we update to go 1.13, 1.4, 1.15 will this script need to be updated?

Reply:
switched to `go mod init` commands to set these up